drivers/cmd/gcloud-spanner/main.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	spannerAdmin "cloud.google.com/go/spanner/admin/database/apiv1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/honeydipper/honeydipper/pkg/dipper"
+	"google.golang.org/api/option"
+	spannerAdminSchema "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+func initFlags() {
+	flag.Usage = func() {
+		fmt.Printf("%s [ -h ] <service name>\n", os.Args[0])
+		fmt.Printf("    This driver supports operator service.\n")
+		fmt.Printf("  This program provides honeydipper with capability of interacting with gcloud spanner\n")
+	}
+}
+
+var driver *dipper.Driver
+var lro *sync.Map
+
+func main() {
+	initFlags()
+	flag.Parse()
+
+	lro = &sync.Map{}
+
+	driver = dipper.NewDriver(os.Args[1], "gcloud-spanner")
+	driver.Commands["backup"] = backup
+	driver.Commands["waitForBackup"] = waitForBackup
+	driver.Run()
+}
+
+func backup(m *dipper.Message) {
+	m = dipper.DeserializePayload(m)
+	params := m.Payload
+	serviceAccountBytes, _ := dipper.GetMapDataStr(params, "service_account")
+	project, ok := dipper.GetMapDataStr(params, "project")
+	if !ok {
+		panic(errors.New("project required"))
+	}
+	instance, ok := dipper.GetMapDataStr(params, "instance")
+	if !ok {
+		panic(errors.New("location required"))
+	}
+	db, ok := dipper.GetMapDataStr(params, "db")
+	if !ok {
+		panic(errors.New("db required"))
+	}
+	timeout := time.Duration(1800)
+	timeoutStr, ok := m.Labels["timeout"]
+	if ok {
+		timeoutInt, _ := strconv.Atoi(timeoutStr)
+		timeout = time.Duration(timeoutInt)
+	}
+
+	var (
+		client *spannerAdmin.DatabaseAdminClient
+		err    error
+	)
+	if len(serviceAccountBytes) > 0 {
+		clientOption := option.WithCredentialsJSON([]byte(serviceAccountBytes))
+		client, err = spannerAdmin.NewDatabaseAdminClient(context.Background(), clientOption)
+	} else {
+		client, err = spannerAdmin.NewDatabaseAdminClient(context.Background())
+	}
+
+	if err != nil {
+		panic(errors.New("unable to create gcloud spanner admin client"))
+	}
+
+	t := time.Now().Add(time.Hour * 24 * 180)
```

Comment:
should the expiration time be a configurable default?

Reply:
Right